sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py

```diff
@@ -78,6 +86,8 @@ def __ne__(self, other: object) -> bool:
 
     def _merge_with(self, other: "ResourceConfiguration") -> None:
         if other:
+            if other.locations:
+                self.locations = other.locations
             if other.instance_count:
                 self.instance_count = other.instance_count
             if other.instance_type:
```

Comment:
Please add some test cases.

Reply:
Removing the changes to locations from this PR.